test/wsdl-test.js

```diff
@@ -5,34 +5,72 @@ var fs = require('fs'),
     assert = require('assert'),
     should = require('should'),
     request = require('request'),
-    http = require('http');
+    http = require('http'),
+    async = require('async');
 
 describe('wsdl-tests', function() {
 
-  describe('wsdl parsing test cases', function () {
+  describe('should parse and describe wsdls under /wsdl/strict dir', function () {
+    var files = [];
 
-    it('should parse and describe - 1 set of wsdls', function (done) {
-      fs.readdirSync(__dirname+'/wsdl/strict').forEach(function(file) {
+    before(function (done) {
+      fs.readdirSync(__dirname + '/wsdl/strict').forEach(function (file) {
         if (!/.wsdl$/.exec(file)) return;
-          soap.createClient(__dirname+'/wsdl/strict/'+file, {strict: true}, function(err, client) {
-            assert.ok(!err);
-            client.describe();
-          });
+        files.push(file);
       });
       done();
     });
 
+    it('should parse and describe wsdls under /wsdl/strict dir', function (done) {
+      async.each(files, function (file, cb) {
+        soap.createClient(__dirname + '/wsdl/strict/' + file, {strict: true}, function (err, client) {
+          assert.ok(!err);
+          client.describe();
+        });
+      });
+      done();
```

Comment:
done is called too early. You need to pass `done` as the last argument to `async.each` so that `done` will be called once `async` finishes all tasks or one of them fails.